speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/BlackBoxData.cpp

```diff
@@ -1,867 +1,151 @@
-    Car.race.curTime = p_race.curTime;
-    Car.race.topSpeed = p_race.topSpeed;
-    Car.race.currentMinSpeedForLap = p_race.currentMinSpeedForLap;
-    Car.race.laps = p_race.laps;
-    Car.race.bestLap = p_race.bestLap;
-    Car.race.nbPitStops = p_race.nbPitStops;
-    Car.race.remainingLaps = p_race.remainingLaps;
-    Car.race.pos = p_race.pos;
-    Car.race.timeBehindLeader = p_race.timeBehindLeader;
-    Car.race.lapsBehindLeader = p_race.lapsBehindLeader;
-    Car.race.timeBehindPrev = p_race.timeBehindPrev;
-    Car.race.timeBeforeNext = p_race.timeBeforeNext;
-    Car.race.distRaced = p_race.distRaced;
-    Car.race.distFromStartLine = p_race.distFromStartLine;
-    Car.race.currentSector = p_race.currentSector;
-    Car.race.nbSectors = p_race.nbSectors;
-    Car.race.trackLength = p_race.trackLength;
-    Car.race.scheduledEventTime = p_race.scheduledEventTime;
-    Car.race.pit = nullptr;  // TODO *pit
-    Car.race.event = p_race.event;
-    // TODO penaltyList
-    Car.race.penaltyTime = p_race.penaltyTime;
-    Car.race.prevFromStartLine = p_race.prevFromStartLine;
-    Car.race.wrongWayTime = p_race.wrongWayTime;
-
-    // Copy p_car.priv
-#define p_priv p_car->priv
-    Car.priv.paramsHandle = nullptr;  // TODO paramsHandle
-    Car.priv.carHandle = nullptr;     // TODO carHandle
-    Car.priv.driverIndex = p_priv.driverIndex;
-    Car.priv.moduleIndex = p_priv.moduleIndex;
-
-    for (int i = 0; i < 4; i++)
-    {
-        Car.priv.modName[i] = p_priv.modName[i];
-    }
-    // TODO wheel
-
-    for (int i = 0; i < 4; i++)
-    {
-        Car.priv.corner[i].x = p_priv.corner[i].x;
-        Car.priv.corner[i].y = p_priv.corner[i].y;
-        Car.priv.corner[i].z = p_priv.corner[i].z;
-        Car.priv.corner[i].xy = p_priv.corner[i].xy;
-        Car.priv.corner[i].ax = p_priv.corner[i].ax;
-        Car.priv.corner[i].ay = p_priv.corner[i].ay;
-        Car.priv.corner[i].az = p_priv.corner[i].az;
-    }
-    Car.priv.gear = p_priv.gear;
-    Car.priv.gearNext = p_priv.gearNext;
-    Car.priv.fuel = p_priv.fuel;
-    Car.priv.fuel_consumption_total = p_priv.fuel_consumption_total;
-    Car.priv.fuel_consumption_instant = p_priv.fuel_consumption_instant;
-    Car.priv.enginerpm = p_priv.enginerpm;
-    Car.priv.enginerpmRedLine = p_priv.enginerpmRedLine;
-    Car.priv.enginerpmMax = p_priv.enginerpmMax;
-    Car.priv.enginerpmMaxTq = p_priv.enginerpmMaxTq;
-    Car.priv.enginerpmMaxPw = p_priv.enginerpmMaxPw;
-    Car.priv.engineMaxTq = p_priv.engineMaxTq;
-    Car.priv.engineMaxPw = p_priv.engineMaxPw;
-
-    for (int i = 0; i < MAX_GEARS; i++)
-    {
-        Car.priv.gearRatio[i] = p_priv.gearRatio[i];
-    }
-    Car.priv.gearNb = p_priv.gearNb;
-    Car.priv.gearOffset = p_priv.gearOffset;
-
-    for (int i = 0; i < 4; i++)
-    {
-        Car.priv.skid[i] = p_priv.skid[i];
+    Car.race = p_car->race;
+    Car.race.bestSplitTime = p_car->race.bestSplitTime ? new double(*p_car->race.bestSplitTime) : nullptr;  // Pointer
```

Comment:
note that this does not copy the entire array and thus the two arrays won't be the same. Furthermore the array pointers are different on the black box side, so copying these arrays is pointless. If you insist on copying these values we will have to look for another way, but for now just keep them at a nullptr

Reply:
Oh those are arrays? Alright will set them to nullptr and adjust the tests accordingly.